homeassistant/components/backup/backup.py

```diff
@@ -0,0 +1,146 @@
+"""Local backup support for Core and Container installations."""
+
+from __future__ import annotations
+
+from dataclasses import asdict, dataclass
+import json
+from pathlib import Path
+import tarfile
+from tarfile import TarError
+from typing import Any, cast
+
+from homeassistant.core import HomeAssistant
+from homeassistant.util.json import json_loads_object
+
+from .agent import BackupAgent, UploadedBackup
+from .const import BUF_SIZE, LOGGER
+from .models import BackupUploadMetadata
+
+
+async def async_get_backup_agents(
+    hass: HomeAssistant,
+    **kwargs: Any,
+) -> list[BackupAgent]:
+    """Register the backup agent."""
+    return [LocalBackupAgent(hass)]
+
+
+@dataclass(slots=True)
+class LocalBackup(UploadedBackup):
+    """Local backup class."""
+
+    path: Path
+
+    def as_dict(self) -> dict:
+        """Return a dict representation of this backup."""
+        return {**asdict(self), "path": self.path.as_posix()}
+
+
+class LocalBackupAgent(BackupAgent):
+    """Define the format that backup agents can have."""
+
+    name = "local"
```

Comment:
In a separate PR, we may want to make name human readable and add a slug-type id.